elements/pfe-navigation/src/pfe-navigation.js

```diff
@@ -0,0 +1,335 @@
+      login: this.shadowRoot.querySelector(".pfe-navigation__main--mobile > pfe-navigation-item:last-of-type")
+    };
+
+    // Initialize active navigation item to empty array
+    this._activeNavigationItems = [];
+    this.overlay = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    // If this element contains light DOM, initialize it
+    if (this.children.length) {
+      // If only one value exists in the array, it starts at that size and goes up
+      this.breakpoints = {
+        main: [0, 1199], // visible from 0 - 1200px
+        search: [768],   // visible from 768px +
+        "mobile-search": [0, 767],
+        language: [768],
+        "mobile-language": [0, 767],
+        login: [768],
+        "mobile-login": [0, 767]
+      };
+
+      // Kick off the initialization of the light DOM elements
+      this._init();
+
+      // Listen for tab events
+      this.addEventListener("keydown", this._keydownHandler);
+
+      // Watch for screen resizing
+      window.addEventListener("resize", this._resizeHandler);
+    } else {
+      console.error("This component does not have any light DOM children.  Please check documentation for requirements.");
+    }
+  }
+
+  disconnectedCallback() {
+    // Remove the scroll, resize, and outside click event listeners
+    window.removeEventListener("resize", this._resizeHandler);
+    window.removeEventListener("scroll", this._stickyHandler);
+    document.removeEventListener("click", this._outsideListener);
+
+    this._observer.disconnect();
+  }
+
+  _resizeHandler(event) {
+    // Set the visibility of items
+    this._setVisibility(this.offsetWidth);
+
+    // Check what the active item is
+    this._activeNavigationItems.forEach(item => {
+      // If the item is open but not visible, update it to hidden
+      if (item.expanded && !item.visible) {
+        item.expanded = false;
+        this._activeNavigationItems = this._activeNavigationItems.filter(i => i !== item);
+      } else if (item.expanded && item.parent && item.parent.visible) {
+        item.parent.expanded = true; // Ensure the parent is open
+        // If the parent item doesn't exist in the active array, add it
+        if (!this._activeNavigationItems.includes(item.parent)) {
+          this._activeNavigationItems.push(item.parent);
+        }
+      }
+    });
+
+    this.overlay = this._activeNavigationItems.length > 0;
+  }
+
+  _stickyHandler() {
+    if(window.pageYOffset >= this.top) {
+      this.classList.add("sticky");
+    } else {
+      this.classList.remove("sticky");
+    }
+  }
+
+  _outsideListener(event) {
+    if ((event.target !== this && event.target.closest("pfe-navigation") === null) || event.path.length > 0 && event.path[0] === this._overlay) {
+      this._activeNavigationItems.map(item => item.close());
+    }
+  }
+
+  _setVisibility(width) {
+    Object.keys(this.breakpoints).forEach(label => {
+      let bps = this.breakpoints[label];
```

Comment:
I think if you use matchMedia with `this.breakpoints`, this could probably be simplified.

Reply:
Cool!  I'd love to learn more about matchMedia for a future update to this component!